tests/PySys/environments/environment_tedge.py

```diff
@@ -0,0 +1,67 @@
+import subprocess
+import time
+
+
+from pysys.basetest import BaseTest
+
+
+class TedgeEnvironment(BaseTest):
+    """Class with helper and convenicence methods for testing tedge"""
+
+    def setup(self):
+        self.sudo = "/usr/bin/sudo"
+        self.tedge = "/usr/bin/tedge"
+        self.tedge_mapper_c8y = "tedge-mapper-c8y"
+        self.sudo = "/usr/bin/sudo"
+        self.systemctl = "/usr/bin/systemctl"
+
+    def wait_if_restarting_mosquitto_too_fast(self):
+        """Make sure we do not restart mosqiotto too fast
+        Systemd will become suspicios when whe restart faster than 5 seconds
+        """
+        minimum_time = 10
+        etimes = subprocess.check_output(
+            "/usr/bin/ps -o etimes $(pidof mosquitto)", shell=True
+        )
+        runtime = int(etimes.split()[1])
+        if runtime <= minimum_time:
+            self.log.info(
+                f"Restarting mosquitto too fast in the last {minimum_time} seconds. It was only up for {runtime} seconds"
+            )
+            # Derive additional delay time and add one safety second
+            delay = minimum_time - runtime + 1
+            self.log.info(f"Delaying execution by {delay} seconds")
+            time.sleep(delay)
+
+    def tedge_connect_c8y(self, expectedExitStatus="==0"):
```

Comment:
This method and subsequent methods are `c8y` specific. So, shouldn't they be moved to `EnvironmentC8y`? And then make `EnvironmentC8y` extend from `TedgeEnvironment`?

Reply:
I added it as separate sub-class, as the EnvironmentC8y automatically connects in the setup method.
The idea was to have a class with helper methods that also make writing tests easier. 
However, your comment also raised some doubts for me regarding naming. I'll at least add some info there.